cscs-checks/mch/automatic_arrays.py

```diff
@@ -0,0 +1,64 @@
+import os
+
+import reframe.utility.sanity as sn
+from reframe.core.pipeline import RegressionTest
+
+
+class AutomaticArraysCheck(RegressionTest):
+    def __init__(self, **kwargs):
+        super().__init__('automatic_arrays_check',
+                         os.path.dirname(__file__), **kwargs)
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
+        self.valid_prog_environs = ['PrgEnv-cray*', 'PrgEnv-pgi*',
+                                    'PrgEnv-gnu']
+        if self.current_system.name in ['daint', 'dom']:
+            self.modules = ['craype-accel-nvidia60']
+            self._pgi_flags = '-acc -ta=tesla:cc60'
+        elif self.current_system.name in ['kesch']:
+            self.modules = ['craype-accel-nvidia35']
+            self._pgi_flags = '-O2 -ta=tesla,cc35,cuda8.0'
+
+        self.num_tasks = 1
+        self.num_gpus_per_node = 1
+        self.num_tasks_per_node = 1
+        self.sourcepath = 'automatic_arrays.f90'
+        self.sanity_patterns = sn.assert_found(r'Result:\s+OK', self.stdout)
+        self.perf_patterns = {
+            'perf': sn.extractsingle(r'Timing:\s+(?P<perf>\S+)',
+                                     self.stdout, 'perf', float)
+        }
+
+        self.aarrays_reference = {
+            'PrgEnv-cray': {
+                'daint:gpu': {'perf': (1.4E-04, None, 0.15)},
+                'dom:gpu': {'perf': (1.4E-04, None, 0.15)},
+                'kesch:cn': {'perf': (2.9E-04, None, 0.15)},
+            },
+            'PrgEnv-gnu': {
+                'daint:gpu': {'perf': (7.4E-03, None, 0.15)},
+                'dom:gpu': {'perf': (7.4E-03, None, 0.15)},
+                'kesch:cn': {'perf': (6.5E-03, None, 0.15)},
+            },
+            'PrgEnv-pgi': {
+                'daint:gpu': {'perf': (1.4E-04, None, 0.15)},
+                'dom:gpu': {'perf': (1.4E-04, None, 0.15)},
+                'kesch:cn': {'perf': (1.4E-04, None, 0.15)},
+            }
+        }
+
+        self.maintainers = ['AJ', 'VK']
```

Comment:
I think you should tag this test as `production`, too.

Reply:
I can do it. However a few checks fail and we will have "red: in the ci.

Reply:
No problem for that. We know that the programming environments are not working properly on Kesch. I will merge it as soon as the rest of the systems are "green".

Reply:
ok